sqlglot/tokens.py

```diff
@@ -1246,3 +1246,45 @@ def _extract_string(self, delimiter: str, escapes=None) -> str:
                 text += self.sql[current : self._current - 1]
 
         return text
+
+    @classmethod
+    def tokenize_native(cls, sql: str) -> t.List[Token]:
+        from sqlglotrs import (  # type: ignore
+            Tokenizer as RsTokenizer,
+            TokenizerSettings as RsTokenizerSettings,
+        )
+
+        settings = RsTokenizerSettings(
+            white_space={k: v.name for k, v in cls.WHITE_SPACE.items()},
+            single_tokens={k: v.name for k, v in cls.SINGLE_TOKENS.items()},
+            keywords={k: v.name for k, v in cls.KEYWORDS.items()},
+            numeric_literals=cls.NUMERIC_LITERALS,
+            identifiers=cls._IDENTIFIERS,
+            identifier_escapes=cls._IDENTIFIER_ESCAPES,
+            string_escapes=cls._STRING_ESCAPES,
+            escape_sequences=cls.ESCAPE_SEQUENCES,  # type: ignore
```

Comment:
Mentioned it in Slack, but leaving this here as well for reference: `cls.ESCAPE_SEQUENCES` and `cls.IDENTIFIERS_CAN_START_WITH_DIGIT` are invalid references after https://github.com/tobymao/sqlglot/commit/be89da3747fa95d98e0d5a28d19d98f5822a8979 because the tokenizer now accesses them through the instance `dialect` attribute, i.e. `self.dialect.ESCAPE_SEQUENCES`.